server/route.go

```diff
@@ -362,6 +362,10 @@ func (c *client) processRouteInfo(info *Info) {
 	s := c.srv
 	remoteID := c.route.remoteID
 
+	c.mu.Unlock()
+	gacc := c.srv.globalAccount()
```

Comment:
I would rather do that on top of function, even if we end-up returning because c.route or c.nc is nil than releasing lock here since technically you would have to re-test c.route and c.nc anyway.
Also, should we get instead the `gacc.sl` under gacc.RLock() and pass that to updateRemoteRoutePerms instead of account?

Reply:
I did in https://github.com/nats-io/gnatsd/pull/849